demo/android/Activity/picovoice-activity-demo-app/src/main/java/ai/picovoice/picovoicedemo/MainActivity.java

```diff
@@ -145,4 +136,28 @@ public void process(View view) {
             displayError("Something went wrong");
         }
     }
+
+    public void showContextCheatSheet(View view) {
+        if (!hasRecordPermission()) {
+            requestRecordPermission();
+            return;
+        }
+
+        AlertDialog.Builder builder = new AlertDialog.Builder(MainActivity.this);
+        ViewGroup viewGroup = findViewById(R.id.content);
+        View dialogView = LayoutInflater.from(view.getContext()).inflate(R.layout.context_cheat_sheet, viewGroup, false);
+        builder.setView(dialogView);
+
+        try {
+            picovoiceManager.start();
```

Comment:
before we call start on PicovoiceManager, you should try to get context information and if it returns an empty string ("") then start the logic for asking for permissions and starting picovoice? 9/10 they'll call context info after a start has already been called